store/postgres/src/writable.rs

```diff
@@ -453,37 +481,27 @@ impl Queue {
                 // these methods would not be able to see that data until
                 // the write transaction commits, causing them to return
                 // incorrect results.
-                let res = match queue.queue.peek().await.as_ref() {
-                    Request::Write {
-                        block_ptr: block_ptr_to,
-                        firehose_cursor,
-                        mods,
-                        data_sources,
-                        deterministic_errors,
-                    } => queue.store.transact_block_operations(
-                        block_ptr_to,
-                        firehose_cursor.as_deref(),
-                        mods,
-                        stopwatch.cheap_clone(),
-                        data_sources,
-                        deterministic_errors,
-                    ),
-                    Request::Revert {
-                        block_ptr,
-                        firehose_cursor,
-                    } => queue
-                        .store
-                        .revert_block_operations(block_ptr.clone(), firehose_cursor.as_deref()),
-                };
+                let req = queue.queue.peek().await;
+                let store = queue.store.cheap_clone();
+                let stopwatch = stopwatch.cheap_clone();
+                let res =
+                    graph::spawn_blocking(
```

Comment:
The naming of the `graph::task_spawn` methods is not very consistent, but you should use `graph::spawn_blocking_allow_panic` here which takes a closure rather than this which takes a future.

Or if you'd rather abort on panic you'd need to add a new helper that takes an `FnOnce` and does that, though catching the panic and propagating as an error seems like what we'd want.

Reply:
I didn't realize that - I just updated the code; now, panics are turned into `StoreError` which seems like a much better way to handle them anyway